### PR TITLE
utils_misc: get_mem_info() api to retrieve meminfo attributes

### PR DESCRIPTION
**Pull request by balamuruhans**
get_mem_info api can be used get memory information from /proc/meminfo
based on attribute regex in host/guest

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -2708,16 +2708,33 @@ def get_free_mem(session, os_type):
...
-        free = re.findall(r"\d+\s\w", output)[0]
+        free = "%s kB" % get_mem_info(session, 'MemFree')
     else:
         output = session.cmd_output("wmic OS get FreePhysicalMemory")
         free = "%sK" % re.findall("\d+", output)[0]
     free = float(normalize_data_size(free, order_magnitude="M"))
     return int(free)
 
 
+def get_mem_info(session=None, attr='MemTotal'):
```
Comment (aliang123) [inline virttest/utils_misc.py]: get_mem_info only applied to linux? And redundant code: "line 2734~2735" vs "line 2715~2716". I think an optimization is needed.

Comment (balamuruhans) [reply virttest/utils_misc.py]: @aliang123 Thank you for reviewing.

> get_mem_info only applied to linux? 

yes, It only applied for Linux, I haven't worked / have Windows env to implement for it.

> And redundant code: "line 2734-2735" with "line 2715-2716". I think an optimization is needed.

As `get_free_mem()` only get's free memory, `get_mem_info()` would help us to retrieve any memory information from /proc/meminfo as it is, ie in kB and `get_free_mem()` should not be disturbed as testcases are using it. so IMHO this 2 line will help us for better compatibility.

Comment (aliang123) [reply virttest/utils_misc.py]: @balamuruhans, got it. It looks good to me.
Acked-by: Aihua Liang<aliang@redhat.com>

Comment (balamuruhans) [reply virttest/utils_misc.py]: @aliang123 Thank you.

**Comment by balamuruhans** [comment]
Tested `get_free_mem()` with and without this patch for ensuring it doesn't regress,
**without this patch**
```
>>> utils_misc.get_free_mem(session, "linux")
100468
```

**with this patch**
```
>>> utils_misc.get_free_mem(session, "linux")
100468
```

```
>>> utils_misc.get_mem_info(session, "MemTotal")
129803584
```

```
\# cat /proc/meminfo | grep 'MemTotal'
MemTotal:       129803584 kB

```

**Comment by balamuruhans** [comment]
Thanks @aliang123, let us merge it.

**Pull request merged**